src/rules/importBlacklistRule.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "import-blacklist",
+        description: Lint.Utils.dedent`
+            Disallows importing the specified modules directly via \`import\` and \`require\`.
+            Instead only sub modules may be imported from that module.`,
+        rationale: Lint.Utils.dedent`
+            Some libraries allow importing their submodules instead of the entire module.
+            This is good practise as it avoids loading unused modules.`,
+        optionsDescription: "A list of blacklisted modules.",
+        options: {
+            type: "array",
+            items: {
+                type: "string",
+            },
+            minLength: 1,
+        },
+        optionExamples: ["true", '[true, "rxjs", "lodash"]'],
+        type: "typescript",
```

Comment:
I don't think this is typescript-only -- I'd throw it in the "functionality" bucket (kind of a catch-all)